hazelcast/src/main/java/com/hazelcast/query/impl/BaseIndexStore.java

```diff
@@ -239,21 +266,39 @@ private Comparable sanitizeScalar(Object input) {
 
     }
 
-    private static class PassThroughFunctor implements CopyFunctor<Data, QueryableEntry> {
+    private class PassThroughFunctor implements CopyFunctor<Data, QueryableEntry> {
 
         @Override
         public Map<Data, QueryableEntry> invoke(Map<Data, QueryableEntry> map) {
+            if (isExpirable()) {
```

Comment:
If i understand this part correctly, end result set, it is the set we return to user, should be the right place to update access times, otherwise it is possible to update access times wrongly. For example `AndPredicate` can contain many predicates but it only cares about smallest result set, if smallest result set has 1 entry and others have 1 million, in this case, we do nearly 1 million access-time updates wrongly where only 1 entry's access-time is needed? Is this correct?

Reply:
This is a good question because it depends on how we define "touch" an entry. If we touch an entry only if it goes to the final result set, we have to make this logic somewhere else. I discovered an issue that when we run a predicate query without indexes we don't update the access time at all (see https://github.com/hazelcast/hazelcast/issues/15137) and so for predicate queries we still have to define "touch" an entry. Now, the touch an entry even if it is part of the intermediate result. The same applies to the HD indexes case.

Reply:
My expectation is updating only final collection of entries, intermediate result `touches` should not be counted as access.

Reply:
I have implemented lazy Record's last time access (LTA) update approach so that we update LTA when we get it through an iterator. We still can touch a bit more intermediate results but in general case we are not able to touch only final results (it will be very expensive) because we may know whether an entry goes to the final result on different member. My solution is a best effort approach which I think is a reasonable approach in this situation.

Reply:
seems better thanks.